solara/hooks/validate_hooks.py

```diff
@@ -0,0 +1,164 @@
+import ast
+import inspect
+import typing as t
+from enum import Enum
+
+DEFAULT_USE_FUNCTIONS = (
+    "use_state",
+    "use_reactive",
+    "use_thread",
+    "use_task",
+    "use_effect",
+    "use_memo",
+)
```

Comment:
I think we can assume all `use_` functions are hooks. Also use hooks have to obey these rules